### PR TITLE
Add action to generate documentation quarto markdown files.

### PR DESCRIPTION
**Pull request by DriesSchaumont**

### CODE REVIEW

File: generate_documentation_qmd/action.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    description: >
+      Jinja template that can be used to customize the resulting qmd files.
+      By default will use the included template.
+  output_dir:
+    required: true
+    description: >
+      Output folder to write to.
+  dest_path:
+    required: false
+    description: Template for output paths
```
Comment (rcannood) [inline generate_documentation_qmd/action.yml:21]: Could you describe what the possible recognized fields are?

File: generate_documentation_qmd/action.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  component_template:
+    required: false
+    description: >
+      Jinja template that can be used to customize the resulting qmd files.
+      By default will use the included template.
+  output_dir:
+    required: true
+    description: >
+      Output folder to write to.
+  dest_path:
```
Comment (rcannood) [inline generate_documentation_qmd/action.yml:19]: `dest_path` is not used any of the steps, or am I mistaken?

File: generate_documentation_qmd/action.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        extra_args=( )
+        if [ ! -z "${{ inputs.component_template }}" ]; then
+          extra_args+="--template ${{ inputs.component_template }}"
+        fi
+
+        viash_tools/target/docker/quarto/generate_documentation_qmd/generate_documentation_qmd \
+          --input "${{ inputs.input_dir }}" "${extra_args[@]}" \
+          --output "${{ inputs.output_dir }}" \
+          --git_repo "${{ github.repository }}" \
+          --git_browse_url "https://github.com/${{ github.repository }}/blob/main/" \
```
Comment (rcannood) [inline generate_documentation_qmd/action.yml:93]: Change to the following?

```bash
          --git_repo "${{ inputs.git_repo }}" \
          --git_browse_url "https://github.com/${{ inputs.git_repo }}/blob/${{ inputs.git_tag }}/" \
```

File: generate_documentation_qmd/action.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      Output folder to write to.
+  dest_path:
+    required: false
+    description: Template for output paths
+    default: "{type}s/{namespace}/{name}.qmd"
+  git_tag:
+    required: false
+    description: >
+      The git tag. Used to create links to the source code in the generated documentation.
+    default: "${{ github.ref_name }}"
```
Comment (rcannood) [inline generate_documentation_qmd/action.yml:27]: Will this default actually work? If not, could do something like:

```bash
git_tag="${{ inputs.git_tag }}"
if [ -z "$git_tag" ]; then
  git_tag="${{ github.ref_name }}"
fi
```

File: generate_documentation_qmd/README.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        ref: "main_build"
+
+    - uses: viash-io/viash-actions/generate_documentation_qmd@generate_documentation_qmd
+      with:
+        input_dir: openpipelines
+        output_dir: openpipelines-bio/website/components
+        dest_path: "{type}s/{namespace}/{name}.qmd"
+        git_repo: openpipelines-bio/openpipelines
+        token: ${{ secrets.GTHB_PAT }}
+        tools_version: main_build
```
Comment (rcannood) [inline generate_documentation_qmd/README.md:81]: Replace with:

```yaml
    steps:
    - uses: viash-io/viash-actions/setup@v2
    - uses: actions/checkout@v3
    - uses: viash-io/viash-actions/generate_documentation_qmd@v2
      with:
        input_dir: src
        output_dir: website/components
        dest_path: "{type}s/{namespace}/{name}.qmd"
        git_repo: openpipelines-bio/openpipelines
        token: ${{ secrets.GTHB_PAT }}
        tools_version: main_build
```

You could definitely add a second example which checks out the website, generates the qmd, creates a pull request to the website.

File: generate_documentation_qmd/README.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+# Example template
+
+````
+
+---
+title: "{{functionality.info.name}}"
+namespace: "{{functionality.info.namespace}}"
+description: "{{functionality.info.short_description}}"
+type: "{{info.type}}"
+toc: false
```
Comment (rcannood) [inline generate_documentation_qmd/README.md:95]: Nice usage of the quadruple backticks, but maybe we can store the template in a separate file and link to it here?

File: generate_documentation_qmd/action.yml (hunk 2 above)
Comment (DriesSchaumont) [reply generate_documentation_qmd/action.yml:19]: Good catch, added!

File: generate_documentation_qmd/action.yml (hunk 3 above)
Comment (DriesSchaumont) [reply generate_documentation_qmd/action.yml]: Changes made

File: generate_documentation_qmd/action.yml (hunk 4 above)
Comment (DriesSchaumont) [reply generate_documentation_qmd/action.yml:29]: This works.

File: generate_documentation_qmd/README.md (hunk 5 above)
Comment (DriesSchaumont) [reply generate_documentation_qmd/README.md]: Adjusted. I will skip adding a second example for now until I have actually build one using this action.

**Review by rcannood — APPROVED** [review]
I think it LGTM!

**Pull request merged**